math/src/main/kotlin/ktx/math/ImmutableVector2.kt

```diff
@@ -0,0 +1,209 @@
+
+    /** @return the distance between this and the other vector */
+    fun dst(otherX: Float, otherY: Float): Float = Vector2.dst(x, y, otherX, otherY)
+
+    /** Returns the result of multiplying this vector by the given matrix */
+    operator fun times(matrix: Matrix3): ImmutableVector2 = ImmutableVector2(
+            x = x * matrix.`val`[0] + y * matrix.`val`[3] + matrix.`val`[6],
+            y = x * matrix.`val`[1] + y * matrix.`val`[4] + matrix.`val`[7]
+    )
+
+    /** Calculates the 2D cross product between this and the given vector */
+    fun crs(otherX: Float, otherY: Float): Float = x * otherY - y * otherX
+
+    /** Returns the angle in radians of this vector relative to the [reference]. Angles are towards the positive y-axis. (typically counter-clockwise) */
+    fun angleRad(reference: ImmutableVector2 = ImmutableVector2.X): Float = angleRad(reference.x, reference.y)
+
+    /** Returns the angle in radians of this vector relative to the reference. Angles are towards the positive y-axis. (typically counter-clockwise) */
+    fun angleRad(otherX: Float, otherY: Float): Float {
+        val result = atan2(y, x) - atan2(otherY, otherX)
+        return when {
+            result > MathUtils.PI -> result - MathUtils.PI2
+            result < -MathUtils.PI -> result + MathUtils.PI2
+            else -> result
+        }
+    }
+
+    /** Returns a vector of same length with the given angle in radians */
+    fun withAngleRad(radians: Float): ImmutableVector2 = ImmutableVector2(len, 0f).withRotationRad(radians)
+
+    /**
+     * Returns a vector of same length rotated by 90 degrees in the given [direction]
+     *
+     * @param direction positive value means toward positive y-axis (typically counter-clockwise). Negative value means toward negative y-axis (typically clockwise).
+     */
+    fun withRotation90(direction: Int): ImmutableVector2 =
+            if (direction >= 0) copy(x = -y, y = x) else copy(x = y, y = -x)
+
+    /** Returns a vector of same length rotated by the given [angle] in radians */
+    fun withRotationRad(angle: Float): ImmutableVector2 {
+        val cos = cos(angle)
+        val sin = sin(angle)
+
+        return ImmutableVector2(
+                x = this.x * cos - this.y * sin,
+                y = this.x * sin + this.y * cos
+        )
+    }
+
+    override fun withLerp(target: ImmutableVector2, alpha: Float): ImmutableVector2 {
+        val invAlpha = 1.0f - alpha
+
+        return ImmutableVector2(
+                x = x * invAlpha + target.x * alpha,
+                y = y * invAlpha + target.y * alpha
+        )
+    }
+
+    override fun withRandomDirection(): ImmutableVector2 = withAngleRad(MathUtils.random(0f, MathUtils.PI2))
+
+    override fun epsilonEquals(other: ImmutableVector2, epsilon: Float): Boolean =
+            epsilonEquals(other.x, other.y, epsilon)
+
+    /** Compares this vector with the other vector, using the supplied epsilon for fuzzy equality testing */
```

Comment:
Would you mind adding parameters documentation? `epsilon` might not be obvious for everyone.